netns_linux.go

```diff
@@ -0,0 +1,27 @@
+//+build linux
+
+package netlink
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+// getThreadNetNS gets the network namespace file descriptor of the thread the current goroutine
+// is running on. Make sure to call runtime.LockOSThread() before this so the goroutine does not
+// get scheduled on another thread in the meantime.
+func getThreadNetNS() (int, error) {
```

Comment:
I think we may want to return `(*os.File, error)` so the file handle can be closed appropriately when no longer needed.

Reply:
Good idea, but namespace handles work a bit differently. The namespace handle we obtain here is not actually created in userspace; its lifecycle is managed by the kernel (created with a syscall). Namespace references in userspace do not count towards the 'refcount', only the amount of threads in the namespace do.

A namespace ceases to exist when the last thread in it exits, therefore I don't believe it makes sense to wrap it into an `os.File`, because the lifecycle of the namespace (and its handle) are not attached to the lifecycle of the `os.File` object at all.